src/pybind/mgr/volumes/fs/operations/versions/metadata_manager.py

```diff
@@ -75,15 +75,18 @@ def flush(self):
 
         fd = None
         try:
-            fd = self.fs.open(self.config_path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
+            tmp_config_path = self.config_path + b'.tmp'
+            fd = self.fs.open(tmp_config_path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
             wrote = 0
             while True:
                 data = conf_data.read()
                 if not len(data):
                     break
                 wrote += self.fs.write(fd, data.encode('utf-8'), -1)
             self.fs.fsync(fd, 0)
-            log.info("wrote {0} bytes to config {1}".format(wrote, self.config_path))
+            log.info("wrote {0} bytes to config {1}".format(wrote, tmp_config_path))
+            self.fs.rename(tmp_config_path, self.config_path)
+            log.info("Renamed {0} to config {1}".format(tmp_config_path, self.config_path))
```

Comment:
please use `f""` format strings if possible
they are easier on the eyes and easier to understand